win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2038,19 +2038,63 @@ HANDLE WINAPI CopyImage(
         return NULL;
     }
 
+    /* If hImage is NULL, then do fast return after setting LastError */
+    if (!hImage)
+    {
+        switch (uType)
+        {
+            case IMAGE_BITMAP:
+                SetLastError(ERROR_INVALID_HANDLE);
+                break;
+            case IMAGE_CURSOR:
+            case IMAGE_ICON:
+                SetLastError(ERROR_INVALID_CURSOR_HANDLE);
+                break;
+            default:
+                SetLastError(ERROR_INVALID_PARAMETER);
+        }
+        return NULL;
+    }
+
     switch(uType)
     {
         case IMAGE_BITMAP:
             return BITMAP_CopyImage(hImage, cxDesired, cyDesired, fuFlags);
         case IMAGE_CURSOR:
         case IMAGE_ICON:
-        /* HACK: Copying bitmaps with LR_COPYFROMRESOURCE flag fails. CORE-17902.
-         * This is a way to return the original bit map if we need
-         * the icons to show up. We need a simpler test. */
         {
             HANDLE handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, fuFlags);
```

Comment:
```suggestion
            HANDLE handle;
            if (!hImage)
            {
                SetLastError(ERROR_INVALID_CURSOR_HANDLE);
                break;
            }
            handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, fuFlags);
```
(Again, I'm wondering whether the NULL-check + last-error setting could be included within the CURSORICON_CopyImagefunction itself. Notice that the CopyIcon() function found later below might benefit from this, if the last-error check turns out to be the same.)

Reply:
Changed. Thanks. For the CURSORICON_CopyImage function comments, maybe this can be considered in another PR.